docs/user-guide/templates.md

```diff
@@ -0,0 +1,64 @@
+# Templates
+
+Templates are snippets of predefined code that people can use to replace a job definition in a [screwdriver.yaml](./configuration/index.md).
```

Comment:
What is a different kind of templates beside `job templates`? I feel like we can combine this and the below section.